### ISSUE TITLE
Feature: Rate Limit

### ISSUE THREAD

**Issue by margau** [franc6/ics_calendar#19]
Hi,
I would propose a rate limit as a new feature.
At the moment, I have the problem, that if getting/parsing the ics causes an error, too many retries are made in a short timeframe.
I just got temporarly blocked from a 3rd party data provider, who complained about 16 requests per minute.

I would suggest something like: "min_interval: 1h" in the YAML-file.

Best regards
margau

**Comment by franc6** [comment]
Can you check if the problem goes away if you add:

```python
    @Throttle(MIN_TIME_BETWEEN_UPDATES)
```

at line 99 of calendar.py?  So it looks like this:

```python
    @Throttle(MIN_TIME_BETWEEN_UPDATES)
    def update(self):
```

From the perspective of HA, a failure shouldn't look any different from success, as the errors shouldn't be passed along.  I think I might've gotten the throttle on the wrong update method. It shouldn't be trying to connect more than once every 15 minutes under any conditions.  If you get a chance to test that before me, and it works, let me know, please.

**Comment by franc6** [comment]
While I haven't been able to reproduce the original problem, I have seen it attempt downloads more frequently than once every 15 minutes.  Unfortunately, "fixing" that causes problems with HA's calendar functionality -- since no data is returned, HA assumes the calendar is empty.  This probably means I'll have to cache the data, and handle the throttling on my own, instead of through HA. :(

@margau Can you give more information on what you were doing when you saw so many requests?  Did you have multiple calendars configured?  What exactly were you doing?  If I can figure out what you were doing, I might be able to reproduce the issue, to be sure I can fully fix it.

**Comment by franc6** [comment]
This should be fixed in 2.1.0

**Comment by franc6** [comment]
If 2.1.0 didn't fix this, 2.9.0 will.
